tools/mo/openvino/tools/mo/moc_frontend/pipeline.py

```diff
@@ -78,6 +78,11 @@ def add_names_to_tensors(model: InputModel, places: List[Place]):
     log.debug('Inputs are same: {}, outputs are same: {}'.format(
         inputs_equal, outputs_equal))
 
+    def create_target_input_shapes(new_input_places):
+        new_input_place_names = [x.get_names()[0] for x in new_input_places]
+        shapes = [shape for shape in argv.placeholder_shapes.values()]
```

Comment:
just one note: I think in the future we need to think of more confident approach for mapping between input names and shapes. Now I am not sure about the order of returned values.
@artkuli, could you please create a ticket to handle this in future for the next release.

Reply:
Yes, sure I will create the separated ticket for it :)